protocols/bftcosi/bftcosi.go

```diff
@@ -295,105 +312,106 @@ func (bft *ProtocolBFTCoSi) handleCommit(comm Commitment) error {
 		log.Lvl4(bft.Name(), "BFTCoSi handle Commit COMMIT")
 	}
 	// set same RoundType as for the received commitment:
-	typedCommitment.TYPE = comm.TYPE
-	typedCommitment.Commitment = commitment
+	typedCommitment := &Commitment{
+		TYPE:       comm.TYPE,
+		Commitment: commitment,
+	}
 	return bft.SendToParent(typedCommitment)
 }
 
-// startPrepareChallenge create the challenge and send its down the tree
-func (bft *ProtocolBFTCoSi) startChallengePrepare() error {
-	// create challenge from message's hash:
-	hash := bft.Suite().Hash()
-	hash.Write(bft.Msg)
-	h := hash.Sum(nil)
+// startChallenge creates the challenge and sends it to its children
+func (bft *ProtocolBFTCoSi) startChallenge(t RoundType) error {
 
-	ch, err := bft.prepare.CreateChallenge(h)
-	if err != nil {
-		return err
-	}
+	log.Lvl4(bft.Name(), "RoundType:", t)
+	if t == RoundPrepare {
+		// need to hash the message before so challenge in both phases are not
+		// the same
+		data := sha512.Sum512(bft.Msg)
```

Comment:
The `crypto/cosi`-signature uses sha512, as far as I can tell. At least replacing that with `suite.Hash` doesn't work...